### ISSUE TITLE
Refined transformations stopped working after 0.7.1 -> 0.7.2

### ISSUE THREAD

**Issue by eugkhp** [scalalandio/chimney#291]
**Checklist**
- [x] I read the documentation at https://scalalandio.github.io/chimney/ and checked that the functionality exists
- [x] I verified that the behavior for my use case doesn't match the documentation
- [x] I checked the https://github.com/scalalandio/chimney/issues and haven't found the issue reported
- [x] I confirmed that the bug is not related to functionality that was deprecated: lifted transformers (`TransformerF`s) or `unsafeOption` flags

**Describe the bug**
After updating minor version from 0.7.1 to 0.7.2 my projected stopped compiling due to:

```
import eu.timepit.refined.types.string.NonEmptyString

case class Bar(address: NonEmptyString)
case class Foo(address: Option[NonEmptyString])

foo.into[Bar].enableOptionDefaultsToNone.tranform
```

```
[error] found   : foo$1.address.value.type (with underlying type String)
[error]  required: T
[error]         from.into[Foo].enableOptionDefaultsToNone.transform,
```

and similar issue for:

```  
import eu.timepit.refined.api.Refined

final case class Tag[U]()
type TaxFeeId = UUID Refined Tag["IdTag"]

case class BarView(
      id: [RefinedId],
)

case class BarView(
      id: Option[RefinedId],
)
```

```
[error]  found   : bar$1.id.value.type (with underlying type java.util.UUID)
[error]  required: T
[error]     def fromBar(fee: Bar): BarView = bar.into[BarView].transform
[error]                                           
```

**Reproduction**
```
import eu.timepit.refined.types.string.NonEmptyString
import io.scalaland.chimney.dsl._

case class Bar(address: NonEmptyString)
case class Foo(address: Option[NonEmptyString])

val bar = Bar(NonEmptyString.unsafeFrom("barToFoo"))

val foo = bar.into[Foo].enableOptionDefaultsToNone.transform
```

https://scastie.scala-lang.org/IcwFIBBzQXCXnPtOEcBT4g

**Which platform do you use**
- [x] JVM
- [x] Scala.js
- [x] Scala Native

**If you checked JVM**
Which JVM distribution do you use (you can provide `java -version` output).
17.0.3

**Comment by krzemin** [comment]
Thanks for reporting! Looks like there was a regression introduced by the combination of #265 and #283.

**Comment by MateuszKubuszok** [comment]
Fixed in [0.7.3](https://github.com/scalalandio/chimney/releases/tag/0.7.3)
